### PR TITLE
Rename `base.py` to `loop.py`

### LINKED ISSUE #11974: Rename `loops/base.py` to `loops/loop.py`
## Proposed refactor

Rename `loops/base.py` to `loops/loop.py`

### Motivation
For consistency, let's conform to the standard of our other main files, like `accelerators/accelerator.py` and `strategies/strategy.py`.

Part of https://github.com/PyTorchLightning/pytorch-lightning/issues/11908

cc @borda @justusschock @awaelchli @rohitgr7 @carmocca @ananthsub @ninginthecloud @daniellepintz

### PR DESCRIPTION
**Pull request by valanm22**
## What does this PR do?

Fixes #11974

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ananthsub — CHANGES_REQUESTED** [review]
Thanks for working on this @valanm22 ! You'll also need to update all of the import paths wherever `pytorch_lightning.loops.base.py` is referenced:
https://github.com/PyTorchLightning/pytorch-lightning/search?q=loops.base

File: docs/source/extensions/loops_advanced.rst [hunk 1]
```diff
@@ -18,7 +18,7 @@ A powerful property of the class-based loop interface is that it can own an inte
 Loop instances can save their state to the checkpoint through corresponding hooks and if implemented accordingly, resume the state of execution at the appropriate place.
 This design is particularly interesting for fault-tolerant training which is an experimental feature released in Lightning v1.5.
 
-The two hooks :meth:`~pytorch_lightning.loops.base.Loop.on_save_checkpoint` and :meth:`~pytorch_lightning.loops.base.Loop.on_load_checkpoint` function very similarly to how LightningModules and Callbacks save and load state.
+The two hooks :meth:`~pytorch_lightning.loops.loop.Loop.on_save_checkpoint` and :meth:`~pytorch_lightning.loops.base.Loop.on_load_checkpoint` function very similarly to how LightningModules and Callbacks save and load state.
```
Comment (akihironitta) [inline docs/source/extensions/loops_advanced.rst:21]:
```suggestion
The two hooks :meth:`~pytorch_lightning.loops.loop.Loop.on_save_checkpoint` and :meth:`~pytorch_lightning.loops.loop.Loop.on_load_checkpoint` function very similarly to how LightningModules and Callbacks save and load state.
```

File: pytorch_lightning/loops/epoch/prediction_epoch_loop.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 import torch
 from deprecate import void
 
-from pytorch_lightning.loops.base import Loop
+from pytorch_lightning.loops.loop import Loop
```
Comment (tchaton) [inline pytorch_lightning/loops/epoch/prediction_epoch_loop.py:7]: Let's add a 1 cycle depreceation. Some projects are relying on the `from pytorch_lightning.loops.base import Loop`

**Comment by daniellepintz** [comment]
Hi @valanm22 are you planning on finishing this PR?

**Comment by valanm22** [comment]
yes, but can I get some instructions on this?

**Comment by daniellepintz** [comment]
sure, could you clarify what exactly you are blocked on?

**Comment by valanm22** [comment]
How to find out the other files that need to be changed?

**Comment by akihironitta** [comment]
Hi @valanm22! Thank you for working on this!

> How to find out the other files that need to be changed?

If you're familiar with a cli tool called `grep`, you can run the following command:
```console
$ grep -rnI "base.loop" .
```

There are also useful tools like the followings:
- https://grep.app/search?q=loops.base&filter[repo][0]=PyTorchLightning/pytorch-lightning
- https://cs.github.com/?scopeName=All+repos&scope=&q=repo%3APyTorchLightning%2Fpytorch-lightning+%22loops.base%22

**Comment by carmocca** [comment]
Closing this one, since you gave the greenlight to another contributor to continue. Thanks for your time!
